File: sample-java/src/main/java/com/ibotta/gradle/aop/java/MainActivity.java
package com.ibotta.gradle.aop.java;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements MessageListener {
    private Button bButtonJava;
    private TextView tvMessage;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bButtonJava = findViewById(R.id.b_button_java);
        tvMessage = findViewById(R.id.tv_message);

        bButtonJava.setOnClickListener(view -> {
            tvMessage.setText("");
            new JavaTargetExample().demonstrateJavaAOP(MainActivity.this);
        });
    }

    @Override
    public void onMessage(String message, CallerType callerType) {
        tvMessage.append(message + "\n");
    }
}

File: sample-mixed/src/main/kotlin/com/ibotta/gradle/aop/mixed/MainActivity.kt
package com.ibotta.gradle.aop.mixed

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.b_button_java
import kotlinx.android.synthetic.main.activity_main.b_button_kotlin
import kotlinx.android.synthetic.main.activity_main.tv_message

class MainActivity : AppCompatActivity(), MessageListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        b_button_kotlin.setOnClickListener {
            tv_message.text = ""
            KotlinTargetExample().demonstrateKotlinAOP(this)
        }

        b_button_java.setOnClickListener {
            tv_message.text = ""
            JavaTargetExample().demonstrateJavaAOP(this)
        }
    }

    override fun onMessage(message: String, callerType: CallerType) {
        tv_message.append("$message\n")
    }
}